File: pdu/python/hako_mavlink_msgs/pdu_cdr_conv_HakoHeartbeat.py
from .pdu_pytype_HakoHeartbeat import HakoHeartbeat
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_HakoHeartbeat(writer: CdrWriter, src: HakoHeartbeat):
    writer.write_uint8(src.type)
    writer.write_uint8(src.autopilot)
    writer.write_uint8(src.base_mode)
    writer.write_uint32(src.custom_mode)
    writer.write_uint8(src.system_status)
    writer.write_uint8(src.mavlink_version)


def cdr_body_to_py_HakoHeartbeat(reader: CdrReader, dst: HakoHeartbeat):
    dst.type = reader.read_uint8()
    dst.autopilot = reader.read_uint8()
    dst.base_mode = reader.read_uint8()
    dst.custom_mode = reader.read_uint32()
    dst.system_status = reader.read_uint8()
    dst.mavlink_version = reader.read_uint8()
    return dst


def py_to_cdr_HakoHeartbeat(src: HakoHeartbeat) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_HakoHeartbeat(writer, src)
    return writer.bytes()


def cdr_to_py_HakoHeartbeat(cdr_payload) -> HakoHeartbeat:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = HakoHeartbeat()
    return cdr_body_to_py_HakoHeartbeat(reader, dst)

File: pdu/javascript/hako_mavlink_msgs/pdu_cdr_conv_HakoHeartbeat.js
import { HakoHeartbeat } from './pdu_jstype_HakoHeartbeat.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduHakoHeartbeatConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { HakoHeartbeat } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint8(src.type);
        writer.write_uint8(src.autopilot);
        writer.write_uint8(src.base_mode);
        writer.write_uint32(src.custom_mode);
        writer.write_uint8(src.system_status);
        writer.write_uint8(src.mavlink_version);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { HakoHeartbeat } dst
     * @returns { HakoHeartbeat }
     */
    static cdr_body_to_js(reader, dst) {
        dst.type = reader.read_uint8();
        dst.autopilot = reader.read_uint8();
        dst.base_mode = reader.read_uint8();
        dst.custom_mode = reader.read_uint32();
        dst.system_status = reader.read_uint8();
        dst.mavlink_version = reader.read_uint8();
        return dst;
    }

    /**
     * @param { HakoHeartbeat } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { HakoHeartbeat }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new HakoHeartbeat());
    }
}
